server/src/service.rs

```diff
@@ -293,4 +320,42 @@ impl Service {
 
         Ok(())
     }
+
+    pub async fn set_locked(
+        &self,
+        locked: bool,
+        objects: &Vec<OwnedObjectPath>,
+    ) -> Result<Vec<OwnedObjectPath>, ServiceError> {
+        let mut locked_changed = Vec::new();
+        let collections = self.collections.lock().await;
+
+        for object in objects {
+            for collection in collections.iter() {
+                if object == collection.path() {
+                    if !collection.is_locked().await {
```

Comment:
shouldn't this compare it to the value of locked passed to it?

Reply:
Yeah, thanks!